client/src/main/java/io/pravega/client/stream/ReaderGroupConfig.java

```diff
@@ -198,8 +228,14 @@ public ReaderGroupConfig build() {
            Preconditions.checkArgument(maxOutstandingCheckpointRequest > 0,
                    "Outstanding checkpoint request should be greater than zero");
 
+           //basic check to make sure autoTruncateAtLastCheckpoint is not true while isSubscriber is false
```

Comment:
This is an incorrect check. "autoTruncateAtLastCheckpoint" can be false for a Subscriber ReaderGroup. In this case, the user needs to use updateTruncationStreamCut API to directly update the truncation Stream-Cut to controller

Reply:
This is to make sure a non subscriber RG does not have `autoTruncateAtLastCheckpoint` set to true. A non subscriber would never even publish checkpoints to the controller so `autoTruncateATLastCheckpoint` should be false. Unless at every use there is going to be a check for whether it is a subscriber and only after that we'll check `autoTruncateAtLastCheckpoint`.

Reply:
Ok.